Remove controller from preset values file

Ranjandas opened the pull request:
Changes proposed in this PR:
- Removed `controller: true` from the preset values file as the controller is now part of connectInject [1697](https://github.com/hashicorp/consul-k8s/pull/1697)


How I've tested this PR:
I built a binary and tried installing using the modified presets.

How I expect reviewers to test this PR:


Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

kschoche reviewed (approved):
Awesome thanks for this, I'd had this fix in the back of my mind and didn't circle back to it yet!
Since your PR didn't run any of the unit tests (because it's a fork PR I'm guessing) it'd be great to confirm that all the tests still pass manually if you have a chance!

Ranjandas commented:
@kschoche, thank you for reviewing. Should we also get rid of `server.replicas: 1` (considering we have one as the default). I left it there so that it is evident to the users that the `server` is installed. Also, which target in the Makefile should I run to run the entire test? Alternatively, should I raise PRs from the source repo than a fork? Having a contribution guideline documented in the repo would have been great.

kschoche commented:
@Ranjandas - Thanks!
I believe that CI only runs the full set of tests against source repo based pull requests (some credential stuff iirc) so its nothing you did incorrectly.
As for also setting `server.replicas:1` I'm on the fence for that, so maybe best to include that in a different PR, simply because I kinda agree that leaving it is somewhat informative for the end user?

To run the tests, I believe you should be okay running `cd cli && go test ./...` to ensure the tests still pass :) 
```
demo $ go test ./...
?       github.com/hashicorp/consul-k8s/cli     [no test files]
ok      github.com/hashicorp/consul-k8s/cli/cmd/install 2.084s
?       github.com/hashicorp/consul-k8s/cli/cmd/proxy   [no test files]
ok      github.com/hashicorp/consul-k8s/cli/cmd/proxy/list      0.921s
ok      github.com/hashicorp/consul-k8s/cli/cmd/proxy/read      1.205s
ok      github.com/hashicorp/consul-k8s/cli/cmd/status  1.103s
ok      github.com/hashicorp/consul-k8s/cli/cmd/uninstall       2.024s
ok      github.com/hashicorp/consul-k8s/cli/cmd/upgrade 1.594s
?       github.com/hashicorp/consul-k8s/cli/cmd/version [no test files]
ok      github.com/hashicorp/consul-k8s/cli/common      10.997s
ok      github.com/hashicorp/consul-k8s/cli/common/flag 1.417s
?       github.com/hashicorp/consul-k8s/cli/common/terminal     [no test files]
?       github.com/hashicorp/consul-k8s/cli/config      [no test files]
ok      github.com/hashicorp/consul-k8s/cli/helm        0.910s
ok      github.com/hashicorp/consul-k8s/cli/preset      1.919s
ok      github.com/hashicorp/consul-k8s/cli/release     0.923s
ok      github.com/hashicorp/consul-k8s/cli/validation  1.491s
?       github.com/hashicorp/consul-k8s/cli/version     [no test files]
```
But you definitely have a valid point about the `CONTRIBUTING.md` file not having instructions for running our unit tests, we should update that guide as its fairly out of date right now in multiple places.

Ranjandas commented:
Thank you so much @kschoche. I have updated the tests, and I have fixed all failures. 

```
ubuntu@dev:~/github.com/ranjandas/consul-k8s/cli$ git rev-parse HEAD
4b39f17d6236fdf094533be1af7997a2eb0acd8d

ubuntu@dev:~/github.com/ranjandas/consul-k8s/cli$ go test ./...
?       github.com/hashicorp/consul-k8s/cli     [no test files]
ok      github.com/hashicorp/consul-k8s/cli/cmd/install (cached)
?       github.com/hashicorp/consul-k8s/cli/cmd/proxy   [no test files]
ok      github.com/hashicorp/consul-k8s/cli/cmd/proxy/list      (cached)
ok      github.com/hashicorp/consul-k8s/cli/cmd/proxy/read      (cached)
ok      github.com/hashicorp/consul-k8s/cli/cmd/status  (cached)
ok      github.com/hashicorp/consul-k8s/cli/cmd/uninstall       (cached)
ok      github.com/hashicorp/consul-k8s/cli/cmd/upgrade (cached)
?       github.com/hashicorp/consul-k8s/cli/cmd/version [no test files]
ok      github.com/hashicorp/consul-k8s/cli/common      (cached)
ok      github.com/hashicorp/consul-k8s/cli/common/flag (cached)
?       github.com/hashicorp/consul-k8s/cli/common/terminal     [no test files]
?       github.com/hashicorp/consul-k8s/cli/config      [no test files]
ok      github.com/hashicorp/consul-k8s/cli/helm        (cached)
ok      github.com/hashicorp/consul-k8s/cli/preset      (cached)
ok      github.com/hashicorp/consul-k8s/cli/release     (cached)
ok      github.com/hashicorp/consul-k8s/cli/validation  (cached)
?       github.com/hashicorp/consul-k8s/cli/version     [no test files]
```

kschoche reviewed (approved):
Most excellent work, thank you!
I noticed we also have some docs cleanup on this front as well to remove the controller references, *if* you have time. Otherwise it's definitely not blocking this PR from going in!
[website/content/docs/k8s/installation/install-cli.mdx](https://developer.hashicorp.com/consul/docs/k8s/installation/install-cli#install-consul-on-kubernetes)

Ranjandas commented:
@kschoche I have updated the doc. Could you please review and help get both PRs merged?
